joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -189,25 +202,35 @@ controller_interface::return_type JointTrajectoryController::update(
   // currently carrying out a trajectory
   if (has_active_trajectory())
   {
+    // Adjust time with scaling factor
+    TimeData time_data;
+    time_data.time = time;
+    rcl_duration_value_t t_period = (time_data.time - time_data_.readFromRT()->time).nanoseconds();
+    time_data.period = rclcpp::Duration::from_nanoseconds(scaling_factor_ * t_period);
+    time_data.uptime = time_data_.readFromRT()->uptime + time_data.period;
+    rclcpp::Time traj_time =
+      time_data_.readFromRT()->uptime + rclcpp::Duration::from_nanoseconds(t_period);
+    time_data_.writeFromNonRT(time_data);
```

Comment:
This should be a writeFromRT, but this doesn't exist see #168 
I'm not sure what is the better way to perform this, but with the set_hold_position I did it that way:
```suggestion    
    time_data_.reset();
    time_data_.initRT(time_data);
```